langgraph/prebuilt/agent_executor.py

```diff
@@ -0,0 +1,123 @@
+import operator
+from typing import Annotated, Sequence, TypedDict, Union
+
+from langchain_core.agents import AgentAction, AgentFinish
+from langchain_core.messages import BaseMessage
+from langchain_core.runnables import RunnableLambda
+
+from langgraph.graph import END, StateGraph
+from langgraph.prebuilt.tool_executor import ToolExecutor
+
+
+def _get_agent_state(input_schema=None):
+    if input_schema is None:
+
+        class AgentState(TypedDict):
+            # The input string
+            input: str
+            # The list of previous messages in the conversation
+            chat_history: Sequence[BaseMessage]
+            # The outcome of a given call to the agent
+            # Needs `None` as a valid type, since this is what this will start as
+            agent_outcome: Union[AgentAction, AgentFinish, None]
+            # List of actions and corresponding observations
+            # Here we annotate this with `operator.add` to indicate that operations to
+            # this state should be ADDED to the existing values (not overwrite it)
+            intermediate_steps: Annotated[list[tuple[AgentAction, str]], operator.add]
```

Comment:
Could we change the observation to `Any` type since the output from tools can be a dict etc. Would be helpful to have an option to stringify it as close to the prompt as possible?